controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			"imageFormat":               "2",
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       r.storageCluster.Namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-rbd-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        r.storageCluster.Namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": r.storageCluster.Namespace,
+		},
+	}
+	storageClassConfiguration = StorageClassConfiguration{
+		storageClass:      storageClass,
+		reconcileStrategy: storagecluster.ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable:           managementSpec.DisableStorageClass,
+		isClusterExternal: r.storageCluster.Spec.ExternalStorage.Enable,
+	}
+	return storageClassConfiguration
+}
+
+func (r *StorageClassClaimReconciler) newCephFilesystemStorageClassConfiguration(storageClassConfiguration StorageClassConfiguration) StorageClassConfiguration {
+	persistentVolumeReclaimDelete := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	managementSpec := r.storageCluster.Spec.ManagedResources.CephFilesystems
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-cephfs", r.storageCluster.Name),
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+		},
+		Provisioner:   fmt.Sprintf("%s.cephfs.csi.ceph.com", r.storageCluster.Namespace),
+		ReclaimPolicy: &persistentVolumeReclaimDelete,
+
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes.
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID": r.storageCluster.Namespace,
+			"fsName":    fmt.Sprintf("%s-cephfilesystem", r.storageCluster.Name),
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       r.storageCluster.Namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-cephfs-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        r.storageCluster.Namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": r.storageCluster.Namespace,
+		},
+	}
+	storageClassConfiguration = StorageClassConfiguration{
+		storageClass:      storageClass,
+		reconcileStrategy: storagecluster.ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable:           managementSpec.DisableStorageClass,
+		isClusterExternal: r.storageCluster.Spec.ExternalStorage.Enable,
+	}
+	return storageClassConfiguration
+}
+
+func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	log := ctrllog.FromContext(r.ctx)
```

Comment:
Why not just create and save the log on the reconciler as `r.log`?

Reply:
Why do we need to do all of that? 
A reconciler instance is a dedicated object for a single request, every request has its own reconciler instance. 
So at the start of the reconciliation (at the head of the `reconcile` func) you do:
```
 r.Log = log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```
then during the reconcile you can just use `r.log`

Of course, if there is a case where we need a more specialized logger we can always create one from the logger that is at `r.log`, but this is the exception and not the common case.